snappy.cc

```diff
@@ -252,6 +279,42 @@ inline char* IncrementalCopy(const char* src, char* op, char* const op_limit,
       if (SNAPPY_PREDICT_TRUE(op >= op_limit)) return op_limit;
     }
     return IncrementalCopySlow(src, op, op_limit);
+
+#elif SNAPPY_HAVE_PWR9
+
+      v16u8 vu8 = vec_xl_len((uint8_t*)src, pattern_size);
+
+      uint32_t np = 16 / pattern_size , ns = pattern_size * np;
+      for (int ix=1; ix<np; ix++)
+          vec_xst_len(vu8, (uint8_t*)&vu8+(ix*pattern_size), pattern_size);
+
+      char* op_end = std::min(op_limit, buf_limit - 15);
+      while (uint64_t(op_limit-op) >= ns) {
+          vec_xst_len(vu8, (uint8_t*)op, ns);
+          op += ns;
+      }
+      if (SNAPPY_PREDICT_TRUE(op >= op_limit)) return op_limit;
+      return IncrementalCopySlow(src, op, op_limit);
+
+#elif SNAPPY_HAVE_VSX
```

Comment:
same for `#elif  defined(__VSX__)`

Reply:
same, nomalized version should be used. @dmiller423 , should be marked as "solved".